cmd/autoheal/object_template.go

```diff
@@ -121,33 +121,55 @@ func (t *ObjectTemplate) Process(object interface{}, data interface{}) error {
 	if glog.V(2) {
 		glog.Infof("Data: %v", data)
 	}
-	return t.processValue(reflect.ValueOf(object), data)
+	v, err := t.processValue(reflect.ValueOf(object), data)
+	if err == nil {
+		object = v.Interface()
+	}
+
+	return err
 }
 
-func (t *ObjectTemplate) processValue(value reflect.Value, data interface{}) error {
-	var err error
-	if value.IsValid() {
-		switch value.Kind() {
+func (t *ObjectTemplate) processValue(inputValue reflect.Value, data interface{}) (val reflect.Value, err error) {
```

Comment:
I am not so happy with what I did here.
However it works, and I couldn't find a better way to do it.

Would appreciate your reviews 
@jhernand @moolitayer @yaacov @nimrodshn

Reply:
sure